### PR TITLE
PAINTROID-304 Fixed text field reset issue

### PR DESCRIPTION
**Pull request by yugalkishore59**
https://jira.catrob.at/browse/PAINTROID-304?jql=project%20%3D%20PAINTROID%20AND%20status%20%3D%20%22Bug%20Backlog%22

Added new variables to store values of current width, height and position of the text field and then assigned these variables back to the text field after creating bitmap in order to restore the last attributes of the text field (in case of font and font effect change).

As mentioned in the above issue (PAINTROID-304), now the dimensions of the text field don't reset when either font or font effect is changed.

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Paintroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [ ] Include reasonable and readable tests verifying the added or changed behaviour
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [ ] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *#paintroid* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: Paintroid/src/main/java/org/catrobat/paintroid/tools/implementation/TextTool.kt [hunk 1]
```diff
@@ -140,25 +140,49 @@ class TextTool(
...
             }
 
             override fun setBold(bold: Boolean) {
                 this@TextTool.bold = bold
                 textPaint.isFakeBoldText = this@TextTool.bold
+                val width = boxWidth
+                val height = boxHeight
+                val position = PointF(toolPosition.x, toolPosition.y)
                 createAndSetBitmap()
+                toolPosition.set(position)
```
[the remark this answers is not in the document]
Comment (ThorstenBandel) [reply Paintroid/src/main/java/org/catrobat/paintroid/tools/implementation/TextTool.kt:183]: And a Test that checks for this change would be needed as well

[the remark this answers is not in the document]
Comment (yugalkishore59) [reply Paintroid/src/main/java/org/catrobat/paintroid/tools/implementation/TextTool.kt]: Now I've removed that extra white space as suggested.

File: Paintroid/src/main/java/org/catrobat/paintroid/tools/implementation/TextTool.kt [hunk 2]
```diff
@@ -140,25 +140,49 @@ class TextTool(
...
             override fun setUnderlined(underlined: Boolean) {
                 this@TextTool.underlined = underlined
                 textPaint.isUnderlineText = this@TextTool.underlined
+                val width = boxWidth
+                val height = boxHeight
+                val position = PointF(toolPosition.x, toolPosition.y)
                 createAndSetBitmap()
+                toolPosition.set(position)
+                boxWidth = width
+                boxHeight = height
```
Comment (CodeChamp-SS) [inline Paintroid/src/main/java/org/catrobat/paintroid/tools/implementation/TextTool.kt:161]: Could you please create a new function for these lines as these same lines are being duplicated in 3-4 methods?

**Review by CodeChamp-SS — CHANGES_REQUESTED** [review]
The changes LGTM and work as expected. 3 of the tests in TextToolIntegrationTest are failing on Jenkins, could you please fix them? Also could you please create a new test for testing the new behavior?

**Comment by yugalkishore59** [comment]
As suggested, now I've created a function named keepAttributeAndcreateBitmap() to minimize the repetition of code lines.

**Comment by CodeChamp-SS** [comment]
> As suggested, now I've created a function named keepAttributeAndcreateBitmap() to minimize the repetition of code lines.

Thanks, the changes LGTM now! And what about the test, are you writing one? Otherwise, I'll approve the changes.

**Comment by yugalkishore59** [comment]
Well, I kinda don't know much about Jenkins. Is there much of a need?

**Comment by CodeChamp-SS** [comment]
> Well, I kinda don't know much about Jenkins. Is there much of a need?
> 
> > > As suggested, now I've created a function named keepAttributeAndcreateBitmap() to minimize the repetition of code lines.
> > 
> > 
> > Thanks, the changes LGTM now! Just noticed a minor static analysis warning on Jenkins, could you please fix that? And what about the test, are you writing one? Otherwise, I'll approve the changes.

yeah, it's a must, it's just one line fix.
image

**Comment by yugalkishore59** [comment]
> > Well, I kinda don't know much about Jenkins. Is there much of a need?
> > > > As suggested, now I've created a function named keepAttributeAndcreateBitmap() to minimize the repetition of code lines.
> > > 
> > > 
> > > Thanks, the changes LGTM now! Just noticed a minor static analysis warning on Jenkins, could you please fix that? And what about the test, are you writing one? Otherwise, I'll approve the changes.
> 
> yeah, it's a must, it's just one line fix. image

Oh, I didn't see that. Is it good now?

**Comment by CodeChamp-SS** [comment]
> > > Well, I kinda don't know much about Jenkins. Is there much of a need?
> > > > > As suggested, now I've created a function named keepAttributeAndcreateBitmap() to minimize the repetition of code lines.
> > > > 
> > > > 
> > > > Thanks, the changes LGTM now! Just noticed a minor static analysis warning on Jenkins, could you please fix that? And what about the test, are you writing one? Otherwise, I'll approve the changes.
> > 
> > 
> > yeah, it's a must, it's just one line fix. image
> 
> Oh, I didn't see that. Is it good now?

yes, thanks :)

**Comment by yugalkishore59** [comment]
You’re welcome. It was the least I could do. :D

**Comment by ThorstenBandel** [comment]
Already fixed in another PR, but thanks for the contribution :)

**Pull request closed without merging**